### PR TITLE
Document release checklist

### LINKED ISSUE #110: Add release process documentation
We need to document the process performed and what tokens to setup for release of the notation CLI. 

/cc @shizhMSFT

### PR DESCRIPTION
**Pull request by shizhMSFT**
Resolve #110

### CODE REVIEW

File: RELEASE_CHECKLIST.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+This document describes the checklist to publish a release via GitHub workflow.
+
+## Release Process
+
+1. Determine a [SemVer2](https://semver.org/)-valid version prefixed with the letter `v` for release. For example, `version="v1.0.0-alpha.1"`.
+2. Bump up the `Version` in [internal/version/version.go](internal/version/version.go#L5) and open a PR for the changes.
+3. Wait for the PR merge.
+4. Generate a GitHub [personal access token (PAT)](https://github.com/settings/tokens/new) with the `repo:public_repo` permission.
+5. In the repository `Settings -> Secrets`, create or update the repository secret `RELEASE_GITHUB_USER_TOKEN` with the PAT generated above.
+6. Make a fresh clone of the repository, check the `git log`, and create a tag by `git tag $version`.
```
[the remark this answers is not in the document]
Comment (SteveLasker) [reply RELEASE_CHECKLIST.md:14]: more info is never bad.

**Review by SteveLasker — APPROVED** [review]
LGTM

**Comment by SteveLasker** [comment]
@iamsamirzon, can you 👀

**Comment by iamsamirzon** [comment]
> @iamsamirzon, can you 👀

LGTM. While obvious, but do we need to add a step in here to open and close an issue to track this release checklist. For example, for alpha-1 release we created three issues one each in notation, notation-go-lib and notary-project repos.

**Comment by SteveLasker** [comment]
Thanks @iamsamirzon , that’s a great item to add for clarity.

**Pull request merged**